core/src/main/scala/com/rackspace/com/papi/components/checker/Validator.scala

```diff
@@ -25,18 +25,27 @@ import javax.xml.transform.dom._
 import javax.xml.transform.sax._
 import javax.xml.transform.stream._
 
+import com.codahale.metrics.RatioGauge.Ratio
+import com.codahale.metrics.{JmxReporter, MetricRegistry, RatioGauge}
 import com.rackspace.com.papi.components.checker.handler.ResultHandler
 import com.rackspace.com.papi.components.checker.servlet._
 import com.rackspace.com.papi.components.checker.step.base.{Step, StepContext}
 import com.rackspace.com.papi.components.checker.step.results.Result
 import com.rackspace.com.papi.components.checker.util.IdentityTransformPool
 import com.rackspace.com.papi.components.checker.wadl.{StepBuilder, WADLDotBuilder}
-import com.yammer.metrics.scala.{Instrumented, Meter, Timer}
-import com.yammer.metrics.util.PercentGauge
+import com.typesafe.scalalogging.slf4j.LazyLogging
+import nl.grons.metrics.scala.{InstrumentedBuilder, Meter, Timer}
 import org.apache.commons.codec.digest.DigestUtils.sha1Hex
 import org.w3c.dom.Document
 
+import scala.util.Try
+
 object Validator {
+  /** The application wide metrics registry. */
+  val metricRegistry = new com.codahale.metrics.MetricRegistry()
```

Comment:
A 1:1 relationship to what? In Repose, for example, you may have multiple validators, but you'll still only be getting (and probably wanting) one metric registry to handle all api-checker metrics. I would expect the metricRegistry to be in an object, just perhaps not this object.

Putting the metricRegistry in the Validator object certainly works. It just seems to me like the function of the metricRegistry is not strictly tied to a Validator. Let me put it this way: if I wanted to add some metric based on a URLFail, would I want to import the Validator object and class? Importing the Validator object and class would provide a lot of code in scope that isn't needed when all I want to do it record some metric. Granted, in this case you can simply mix-in the Instrumented trait, which is great for handling this type of thing. Still, I would personally break out the metricRegistry for the sake of clarity and code organization, but I understand that it will work as-is. As this is @RackerWilliams brainchild, I defer to him.